File: packages/http-client-java/generator/http-client-generator-test/src/test/java/parameters/bodyroot/BodyRootTests.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package parameters.bodyroot;

import org.junit.jupiter.api.Test;
import parameters.bodyroot.models.BodyRootModel;

public class BodyRootTests {

    private final BodyRootClient client = new BodyRootClientBuilder().buildClient();

    @Test
    public void testNested() {
        client.nested(new BodyRootModel().setCategory("widget").setLinkType("hard").setWasSuccessful(true));
    }
}

File: packages/http-client-csharp/generator/TestProjects/Spector.Tests/Http/Parameters/BodyRoot/BodyRootTests.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Threading.Tasks;
using NUnit.Framework;
using Parameters.BodyRoot;

namespace TestProjects.Spector.Tests.Http.Parameters.BodyRoot
{
    public class BodyRootTests : SpectorTestBase
    {
        [SpectorTest]
        public Task Nested() => Test(async (host) =>
        {
            var bodyRoot = new BodyRootModel
            {
                Category = "widget",
                LinkType = "hard",
                WasSuccessful = true
            };
            var response = await new BodyRootClient(host, null).NestedAsync(new NestedParameterBody(bodyRoot));
            Assert.AreEqual(204, response.GetRawResponse().Status);
        });
    }
}
